internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -4,101 +4,122 @@ import (
 	"context"
 	"time"
 
+	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 	"golang.org/x/sync/errgroup"
 
-	"github.com/G-Research/armada/internal/jobservice/configuration"
+	"github.com/G-Research/armada/internal/jobservice/events"
 	"github.com/G-Research/armada/internal/jobservice/repository"
 	"github.com/G-Research/armada/pkg/api"
-	"github.com/G-Research/armada/pkg/client"
 )
 
 // Service that subscribes to events and stores JobStatus in the repository.
 type EventsToJobService struct {
 	queue                string
 	jobSetId             string
 	jobId                string
-	jobServiceConfig     *configuration.JobServiceConfiguration
-	jobServiceRepository repository.SQLJobService
+	eventClient          events.JobEventReader
+	jobServiceRepository repository.JobTableUpdater
 }
 
 func NewEventsToJobService(
 	queue string,
 	jobSetId string,
 	jobId string,
-	jobServiceConfig *configuration.JobServiceConfiguration,
-	jobServiceRepository repository.SQLJobService,
+	eventClient events.JobEventReader,
+	jobServiceRepository repository.JobTableUpdater,
 ) *EventsToJobService {
 	return &EventsToJobService{
 		queue:                queue,
 		jobSetId:             jobSetId,
 		jobId:                jobId,
-		jobServiceConfig:     jobServiceConfig,
+		eventClient:          eventClient,
 		jobServiceRepository: jobServiceRepository,
 	}
 }
 
-// Subscribes to a JobSet from jobsetid
-func (eventToJobService *EventsToJobService) SubscribeToJobSetId(context context.Context) error {
-	return eventToJobService.streamCommon(&eventToJobService.jobServiceConfig.ApiConnection, context)
+// Subscribes to a JobSet from jobsetid. Will retry until there is a successful exit, up to the TTL
+func (eventToJobService *EventsToJobService) SubscribeToJobSetId(context context.Context, ttlSecs int64) error {
```

Comment:
I don't know if we want the same config as subscribe job set as the time to determine if our connection is bad.  

Should we just create a new config for TTL?

Reply:
That sounds reasonable, but as a matter of course maybe it doesn't matter? If the subscribe job set time ttl expired, this will quit retrying. If this were larger than the subscribe time, it would continue trying but with certain failure.

Reply:
Yea I see your point.  But the SubscribeJobSetTime can be as long as how long it takes for a job-set to run.  So we were thinking that it should be close to the lifetime of a running job.  On the order of hours/days maybe.  

So if we keep the same config, we will be retrying connections for that long of time before we decide it is a failure.  

Right now, our SubscribeJobSetTime is 10000 seconds.

Reply:
I probably don't understand it well enough, but if the job is still active we should still attempt to subscribe, right?

Reply:
Ideally but I don't think we have too.  

We unsubscribe if a job state doesn't get updated after x amount of time.  If a job is running for 2-3 weeks, we can unsubscribe and resubscribe without any issue.  All unsubscribed means is that we no longer have a connection to the job set.  It doesn't stop the job or anything like that.

Reply:
this retry loop will exit when IsJobSetSubscribed() returns false.

Reply:
So in this PR, we will attempt to retry for the duration of the SubscribeJobSetTime if we're getting errors from one of the go routines. This is the same amount of time that it would be subscribed. Once the subscribed jobs cleanup occurs, and the job is unsubscribed, this loop exits.